packages/presentation/src/document/PresentationDocumentProvider.tsx

```diff
@@ -49,7 +46,10 @@ export function PresentationDocumentProvider(props: {
     [optionsArray, parent, register],
   )
 
-  useLayoutEffect(() => registerRef.current(options), [options])
+  const registerEffectEvent = useEffectEvent((options: PresentationPluginOptions) =>
+    register(options),
+  )
+  useLayoutEffect(() => registerEffectEvent(options), [registerEffectEvent, options])
```

Comment:
What `useEffectEvent` does internally, is similar to the original logic:
```ts
  const registerRef = useRef(register)
  registerRef.current = register
```
But it's accomplishing this in a way that's Concurrent React safe, and memoization safe.